File: 3636-check-balanced-string/3636-check-balanced-string.cpp
class Solution {
public:
    bool isBalanced(string num) {
        int even = 0, odd = 0;
        for (int i = 0; i < num.length(); ++i) {
            int sum = num[i] - '0';
            if (i & 1)
                even += sum;
            else
                odd += sum;
        }
        return even == odd;
    }
};

File: 3636-check-balanced-string/3636-check-balanced-string.py
class Solution:
    def isBalanced(self, num: str) -> bool:
        e, o = 0, 0
        for i in range(len(num)):
            dig = ord(num[i]) - ord('0')
            if i & 1:
                e += dig
            else:
                o += dig
        
        return e == o